src/lambda_codebase/initial_commit/bootstrap_repository/adf-build/shared/helpers/terraform/get_accounts.py

```diff
@@ -0,0 +1,136 @@
+import boto3
+import json
+import os
+import logging
+from paginator import paginator
+
+# Configure logging
+logging.basicConfig(level=logging.INFO)
+LOGGER = logging.getLogger(__name__)
+LOGGER.setLevel(logging.INFO)
+
+MANAGEMENT_ACCOUNT_ID = os.environ["MANAGEMENT_ACCOUNT_ID"]
+TARGET_OUS = os.environ.get("TARGET_OUS")
+sts = boto3.client('sts')
+
+
+def main():
+    accounts = get_accounts()
+    with open('accounts.json', 'w') as outfile:
+        json.dump(accounts, outfile)
+
+    if TARGET_OUS:
+        accounts_from_ous = get_accounts_from_ous()
+        with open('accounts_from_ous.json', 'w') as outfile:
+            json.dump(accounts_from_ous, outfile)
+
+
+def list_organizational_units_for_parent(parent_ou):
+    organizations = get_boto3_client('organizations', f'arn:aws:sts::{MANAGEMENT_ACCOUNT_ID}:role/OrganizationAccountAccessRole-readonly', 'getOrganizationUnits')
+    organizational_units = [
+        ou
+        for org_units in organizations.get_paginator("list_organizational_units_for_parent").paginate(ParentId=parent_ou)
+        for ou in org_units['OrganizationalUnits']
+    ]
+    return organizational_units
+
+
+def get_accounts():
+    # Return an array of objects like this: [{'AccountId':'xxx','Email':''}]
+    account_details = []
+    LOGGER.info(
+        "Management Account ID: %s",
+        MANAGEMENT_ACCOUNT_ID
+    )
+    # Assume a role into the management accounts role to get account ID's and emails
+    organizations = get_boto3_client('organizations', f'arn:aws:sts::{MANAGEMENT_ACCOUNT_ID}:role/OrganizationAccountAccessRole-readonly', 'getaccountIDs')
```

Comment:
Same here, could you add support for the correct partition (:aws:) part?

Reply:
added support to partition